### DISCUSSION TITLE
Unable to publish a contract on Sui Devnet

### DISCUSSION THREAD

**Discussion by Tanmay53** [MystenLabs/sui#13311] [Q&A]
I am trying to run the following command to publish the contracts on sui Devnet

```bash
sui client publish sources/nft_example.move --gas-budget 20000
```

This result into the following error

```
RPC call failed: ErrorObject { code: InvalidParams, message: "Transaction has non recoverable errors from at least 1/3 of validators", data: Some(RawValue([[{"UserInputError":{"error":{"GasBudgetTooLow":{"gas_budget":2000,"min_budget":1000000}}}},5000,["p3Orq1DtDFLufr2VgRDX6ldjYU0SaM68J1zDT5uwZFYQ0fro82cDPXJHDl0Y7EMXBDnt0KeO7cHyCmS1wx2INRYk2zK6yP+ctb+fT0rffKeD18mhTAaCtua8J8Er59r1","r2FBEItU3hZi/JqPTpWEsFXQK1BTkhzCrOTUIgDYfZZQJBPO6yrgNMeCqH5z5n+1AUCEYQlbXaiENLtwM5dzGhTuzzfutEP66OKcyMMrEvs6YyB03m8NsvtTRqdpqETF"]]])) }

Caused by:
    RPC call failed: ErrorObject { code: InvalidParams, message: "Transaction has non recoverable errors from at least 1/3 of validators", data: Some(RawValue([[{"UserInputError":{"error":{"GasBudgetTooLow":{"gas_budget":2000,"min_budget":1000000}}}},5000,["p3Orq1DtDFLufr2VgRDX6ldjYU0SaM68J1zDT5uwZFYQ0fro82cDPXJHDl0Y7EMXBDnt0KeO7cHyCmS1wx2INRYk2zK6yP+ctb+fT0rffKeD18mhTAaCtua8J8Er59r1","r2FBEItU3hZi/JqPTpWEsFXQK1BTkhzCrOTUIgDYfZZQJBPO6yrgNMeCqH5z5n+1AUCEYQlbXaiENLtwM5dzGhTuzzfutEP66OKcyMMrEvs6YyB03m8NsvtTRqdpqETF"]]])) }
```

The wallet balance is **10 SUI**.

Code Reference : https://github.com/sui-foundation/encode-sui-educate/blob/main/lesson-3/sources/onchain_identity.move

**Comment by OPTIMUS-PX** [comment]
Help me to know how

**Reply by Tanmay53** [reply]
?

**Comment by OPTIMUS-PX** [comment]
Generate public key

**Reply by Tanmay53** [reply]
Already done with the help of `sui client addresses`

**Comment by hwerbi** [comment]
hi 
did you try "sui move build " the package before the publish, 
also, increase the  gas budget

**Reply by Tanmay53** [reply]
Yes, I ran `sui move build` before publishing the package.

When I increase the gas budget to 1,000,000 as stated in the error, I get the following error though I have 10 sui in the Devnet account.

![Screenshot 2023-08-08 144540](https://github.com/MystenLabs/sui/assets/52573153/045138d1-615a-4bf4-b772-fc90c2e7b627)

**Comment by hwerbi** [comment]
You can verify the gas using sui client objects

You can try to increase more the gas
Or
Try to publish the hole folder you build not just the module

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Tanmay53** [reply]
I tried. Still the same issue.

**Comment by hwerbi** [comment]
I just tried to publish a package and I got the same error I  fixed it as indicated in the figure. So I suggest:

  *   publish a folder, not a file
  *   be inside the folder to publish
  *   indicate the whole path to the folder
  *   increase the gas to 1,000,000,000

I hope this helps you

________________________________
De : Tanmay Khatri ***@***.***>
Envoyé : mercredi 9 août 2023 16:57
À : MystenLabs/sui ***@***.***>
Cc : hwerbi ***@***.***>; Comment ***@***.***>
Objet : Re: [MystenLabs/sui] Unable to publish a contract on Sui Devnet (Discussion #13311)


I tried. Still the same issue.

**Reply by Tanmay53** [reply]
Thanks, I will git it a try.
